src/sentry/api/bases/organization_events.py

```diff
@@ -33,13 +34,34 @@ def has_feature(self, organization, request):
 
     def get_snuba_filter(self, request, organization, params=None):
         if params is None:
-            params = self.get_filter_params(request, organization)
+            params = self.get_snuba_params(request, organization)
         query = request.GET.get("query")
         try:
             return get_filter(query, params)
         except InvalidSearchQuery as e:
             raise ParseError(detail=six.text_type(e))
 
+    def get_snuba_params(self, request, organization):
+        with sentry_sdk.start_span(op="discover.endpoint", description="filter_params") as span:
+            if len(request.GET.getlist("field")) > MAX_FIELDS:
+                raise ParseError(
+                    detail="You can view up to {0} fields at a time. Please delete some and try again.".format(
+                        MAX_FIELDS
+                    )
+                )
+
+            span.set_tag("organization", organization)
+            params = self.get_filter_params(request, organization)
+            params = self.quantize_date_params(request, params)
+
+            has_global_views = features.has(
+                "organizations:global-views", organization, actor=request.user
+            )
+            if not has_global_views and len(params.get("project_id", [])) > 1:
+                raise ParseError(detail="You cannot view events from multiple projects.")
```

Comment:
I know this is a refactor but shouldn't this be a 404?

Reply:
🤔 should it be a 404? At this point we've checked that the user can use the endpoint that we're getting params for, but what they're asking for is invalid

Reply:
I'd argue that this is a permission error as they don't have the right features enabled to do this. The following snippet is in many parts of the code base

```python
if not self.has_feature(organization, request):
    return Response(status=404) 
```

And I'd argue the error in the two cases should at least be the same. Though I do see what you mean.

Reply:
@Zylphrex Hmm, I think that's a reasonable argument too. But to counter that, other parts of the app 400 on this error too (eg. issues). And I think part of the reason for that is that with a 400 we can include the detail to give a helpful error message.